### PR TITLE
Microverse NYT Article

### PR DESCRIPTION
**Pull request by AndreasGalster**
Hi dear microverse code reviewer :).

A few notes on implementation:
- This pull request is for the NYT article.
- We've heavily used grid for the layout of the page.
- We load individual styles via `@import`
  - We know this is bad practice in normal css but in a production project this would be a @import with .scss and would bundle it all up into one file
- All linting tests are passing

### CODE REVIEW

**Review by jubaan — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle:

## :wave: :wave: Hello Team!! Great effort on your project!!

:100: Using semantic HTML
:100: The project looks awesome! almost like the original! :clap: :+1: 
:100: No inline comments
:100: Separate CSS in smaller parts. (You're right not best practice with css but with scss another story :+1: ).

Nice! :sunglasses:  The project has a great solid start. We should know focus on improving it.

## Required Changes
### [Mandatory] :red_circle: 

- [ ] README refers to an unrelated project.
image

- [ ] README is not descriptive enough. We should be more specific about the description of the project the main goal and how the team tackle it.

- [ ] Getting Started section, move the pre-requisites first, and then the steps to have a local version of the project. Let's add some bullet points to make the steps more visual.

- [ ] NYT Logo is missing
image

- [ ] The project is missing sections.  **spirals in the sky**, **a special time**

- [ ] The project is missing embedded video.
image

- [ ] Missing embedded image
image

- [ ] Images should be also links
image

- [ ] Titles and images should be links
image

- [ ] The NYT logo should be a link
image

- [ ] Add unnecessary files to ```.gitignore``` or remove them
image

### [OPTIONAL]  :large_blue_circle: 

- [ ] Prefer specific classes over general CSS selectors. Example:
Actual: ```button```
Improved :+1: : ```.navbar__button```

- [ ] Prefer CSS **classes** over CSS **ids**. Example:
Actual: ```#title-area```
Improved :+1: : ```.title-area```

- [ ] The ```cursor: pointer```would not be necessary once you convert the elements to links. When this is done please remove it.

### [Comments] :green_circle: 
- The README is one of the most important files in your repository; it is the first thing a reader/recruiter while wandering through your repositories. 
- Only use IDs when you're sure is a unique element in the project, that there are no other elements that could be similar, and that is not going to be reused somewhere else. 
- To understand better what is happening visually in a Website, use the browser Inspector Mode. This will help to mimic not only the appearance but some of the functionality that it has, like like links and images that act as links.

## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

**Comment by AndreasGalster** [comment]
Hi @jubaan, thanks for the review. We'll work on this tomorrow.

> * [ ]  Add unnecessary files to `.gitignore` or remove them
>   image

Regarding this comment: I think we should [keep the lock file](https://stackoverflow.com/questions/44552348/should-i-commit-yarn-lock-and-package-lock-json-files/44904494#:~:text=You%20should%20commit%201%20dependency,build%20%2B%20develop%20a%20project%20with.&text=lock%20should%20be%20committed%2C%20if,you%20commit%20both%20the%20yarn.) and the prettier config (prettier needs a config so it auto-formats, even if the file is empty).

Makes sense to remove the `node_modules` folder from the repo :)

**Comment by jubaan** [comment]
> Hi @jubaan, thanks for the review. We'll work on this tomorrow.
> 
> > * [ ]   Add unnecessary files to `.gitignore` or remove them
> >   image
> 
> Regarding this comment: I think we should [keep the lock file](https://stackoverflow.com/questions/44552348/should-i-commit-yarn-lock-and-package-lock-json-files/44904494#:~:text=You%20should%20commit%201%20dependency,build%20%2B%20develop%20a%20project%20with.&text=lock%20should%20be%20committed%2C%20if,you%20commit%20both%20the%20yarn.) and the prettier config (prettier needs a config so it auto-formats, even if the file is empty).
> 
> Makes sense to remove the `node_modules` folder from the repo :)

Good catch!. Still, adding them to ```.gitignore``` will fix this issue. This will help you have locally all the files needed to work with the project (like the prettier files) and in github only the files related without the extra ones. You kill two birds with the same stone :wink: . Give it a try :smiley:

**Comment by AndreasGalster** [comment]
Hi @jubaan we have made the required changes. I've added the `node_modules` folder to the .gitignore and removed it from the branch.

Please note I've fixed the missing NYT logo in the header. It was due to a wrong path `/assets/...jpg` -> `./assets/...jpg`. It's not yet showing up on the live version on raw.githack since the `index.html` file is still cached.

**Comment by umairarshadbutt** [comment]
##
## Pull Request under review by Umair Arshad (TSE)

File: styles/content.css [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+#foot-note p,
+#foot-note a {
+  font-size: 12px;
+  color: var(--nyt-grey-text);
+  font-family: var(--sans-font);
+}
+
+.wide-content {
+  max-width: 945px;
+  margin-top: 55px !important;
```
Comment (umairarshadbutt) [inline styles/content.css]: Remove the `!important`

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi 
Good job with the changes! :tada: We are almost there.
We, however, do have some few changes to carry out before we proceed (see review comment).

##

 For this project, we have to make sure nothing is messed up. Kindly work on the suggested changes  👇.
- [ ] Remove `!important `from css file. ([Article for support](https://speckyboy.com/good-bad-css-practices/)) ( [screenshot](https://user-images.githubusercontent.com/29689273/93404399-8997b500-f8a3-11ea-8f88-fb12b8f6ca80.png) from Pathwright)
- [ ]  Please follow the instructions about .gitignore as previous TSE suggested

##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you stick at any point or have any questions, let me know on slack at any time. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Comment by AndreasGalster** [comment]
Hi @umairarshadbutt can you please explain what exactly we should do with `.gitignore`?

Not adding the lock file or the prettier-config to the repo (so adding it to the .gitignore) is problematic in collaborative workflows, so that's why I only added the `node_modules` folder to the .gitignore

- The lock-file is required to be committed to the repo to ensure everyone on the project has the same dependency resolutions. The only reason why lock-files by yarn/npm were invented was exactly for this purpose, to commit them to the repo 😄 
- The prettier file is required to be committed to the repo so everyone has the same prettier config. If we'd update our prettier file, then the other collaborators on the project wouldn't get the same prettier config if it's in `.gitignore`. That would be a problem for a real project, wouldn't it? It's the same as if we'd add the lining configs to our .gitignore, in which case linting would be broken, right :)?

**Comment by umairarshadbutt** [comment]
Hi @AndreasGalster 
 Files not related to the project should not be committed and added to .gitignore (e.g. .DS_Store) ([screenshot](https://user-images.githubusercontent.com/29689273/94254268-fc59fd80-ff3f-11ea-9f59-c5c13ca39d24.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**Comment by umairarshadbutt** [comment]
`yarn.lock` is used for top-level projects where `yarn.lock` files will not generate unnecessary duplicates. So you can not use it in your project.

**Comment by AndreasGalster** [comment]
Thank you for your feedback, we've made the changes requested :)

**Review by umairarshadbutt — APPROVED** [review]
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
### The [clone](https://raw.githack.com/AndreasGalster/mv-nyt/article/index.html) shows a good understanding and usage of Bootstrap. :100:
##
- The cloned page is rendering as it should and meets all the project specifications.
- Pull Request has a meaningful title and a short description.
- CSS is clean. No linter errors at this point and no bad practices used as well.
- The markup as well has no errors nor warnings to show. Good job with that.
- The repo has followed all the guidelines there are to follow (GitHub flow etc.)
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.**
- Congratulations on the splendid job @AndreasGalster and @yocosaka! :+1: 
- You can now merge to master and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal:
##
**If you have any questions, let me know on slack at any time. I will be happy to help.**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Pull request merged**